File: src/main/java/com/example/hnefatafl/repository/GameRepository.java
package com.example.hnefatafl.repository;

import com.example.hnefatafl.model.MongoGame;
import org.springframework.data.mongodb.repository.MongoRepository;

import java.util.List;

public interface GameRepository extends MongoRepository<MongoGame, String> {
    List<MongoGame> findByType(String type);
}

File: src/main/java/com/example/hnefatafl/repository/GameRepository.kt
package com.example.hnefatafl.repository

import com.example.hnefatafl.model.MongoGame
import org.springframework.data.mongodb.repository.MongoRepository

interface GameRepository : MongoRepository<MongoGame?, String?> {
    fun findByType(type: String?): List<MongoGame?>?
}